plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2634,10 +2634,14 @@ private static void updatePartitioning(Table icebergTable, Transaction transacti
         else {
             PartitionSpec partitionSpec = parsePartitionFields(schema, partitionColumns);
             Set<PartitionField> partitionFields = ImmutableSet.copyOf(partitionSpec.fields());
-            difference(existingPartitionFields, partitionFields).stream()
+
```

Comment:
> Fix alter partition columns in iceberg

Could you explain the details in a commit body?

Reply:
added explanation in the commit body.